### PR TITLE
5001 create organization over quota email

### LINKED ISSUE #5001: Create 'Organization over quota' email
When a multiuser organization is over MB quota, we need to send an email to the ORG admin

### PR DESCRIPTION
**Pull request by juanignaciosl**
This closes #5001. @matallo , @ethervoid could you CR this, please?

@brittanymicek, @csobier this is the mail screenshot including the subject:

screen shot 2015-09-14 at 13 50 28

### CODE REVIEW

**Comment by juanignaciosl** [comment]
Retest this, please.

File: app/mailers/organization_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class OrganizationMailer < ActionMailer::Base
+  default from: "cartodb.com <support@cartodb.com>"
+  layout 'mail'
+
+  def quota_limit(organization)
+    @organization = organization
+    @subject = "Your organization #{@organization.name} has reached its quota"
+    @link = "mailto:support@cartodb.com"
+
+    mail to: @organization.owner.email,
```
Comment (brittanymicek) [inline app/mailers/organization_mailer.rb]: Looks good! Thanks team!

On Mon, Sep 14, 2015 at 9:05 AM, Hound <notifications@github.com> wrote:

> In app/mailers/organization_mailer.rb
> <https://github.com/CartoDB/cartodb/pull/5499#discussion_r39390499>:
>
> > @@ -0,0 +1,14 @@
> > +class OrganizationMailer < ActionMailer::Base
> > +  default from: "cartodb.com <support@cartodb.com>"
> > +  layout 'mail'
> > +
> > +  def quota_limit(organization)
> > +    @organization = organization
> > +    @subject = "Your organization #{@organization.name} has reached its quota"
> > +    @link = "mailto:support@cartodb.com"
> > +
> > +    mail to: @organization.owner.email,
>
> Trailing whitespace detected.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CartoDB/cartodb/pull/5499/files#r39390499>.
>



-- 
*Brittany Micek*
Content Writer

*CartoDB—Map Your World's Data*
143 Roebling St 2A, New York, 11211, USA
T: + <%2B34695497365>19174633232
W: www.cartodb.com

Live sessions to learn CartoDB <http://cartodb.com/webinars/>

File: app/models/organization.rb [hunk 2]
```diff
@@ -229,8 +229,17 @@ def assigned_seats
...
   end
 
+  def notify_if_disk_quota_limit
+    ::Resque.enqueue(::Resque::OrganizationJobs::Mail::QuotaLimit, self.id) if disk_quota_limit
+  end
+
   private
 
+  # Returns true if disk quota won't allow new signups with existing defaults
+  def disk_quota_limit
```
Comment (ethervoid) [inline app/models/organization.rb]: Maybe I should call this `disk_quote_limit_reached` or something like that. This way it seems like the method is going to return the disk quota limit quantity.

File: app/mailers/organization_mailer.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+class OrganizationMailer < ActionMailer::Base
+  default from: "cartodb.com <support@cartodb.com>"
+  layout 'mail'
+
+  def quota_limit(organization)
```
Comment (ethervoid) [inline app/mailers/organization_mailer.rb]: Same here `quota_limit_reached` for example

File: lib/resque/user_jobs.rb [hunk 4]
```diff
@@ -4,6 +4,24 @@
 require_relative '../cartodb/metrics'
 
 module Resque
+
+  module OrganizationJobs
+
+    module Mail
+
+      module QuotaLimit
```
Comment (ethervoid) [inline lib/resque/user_jobs.rb]: `QuotaLimitReached` or even `UserDiskSpaceQuotaLimitReached` because in the future could be other quotas

**Comment by ethervoid** [comment]
Despite some minot tweaks, its good for me

File: app/models/organization.rb (hunk 2 above)
Comment (juanignaciosl) [inline app/models/organization.rb]: Right, changed to `disk_quota_limit_reached?´

File: app/mailers/organization_mailer.rb (hunk 3 above)
Comment (juanignaciosl) [inline app/mailers/organization_mailer.rb]: Sure. `quota_limit_reached`

File: lib/resque/user_jobs.rb (hunk 4 above)
Comment (juanignaciosl) [inline lib/resque/user_jobs.rb]: :+1: `DiskQuotaLimitReached`

**Comment by juanignaciosl** [comment]
@matallo don't forget CR-ing this when you have an slot, please ;-). I've just forwarded you the triggered email, and it's currently deployed at ded04 if you want to try. Organization "juanignaciosl-org-ded-04" will trigger the email if you sign up with another user (although you should change juanignaciosl-org-ded-04-admin email, of course).

**Comment by matallo** [comment]
screen shot 2015-09-15 at 12 54 11

updated the email template, how this is looking @juanignaciosl @josecruz

**Comment by juanignaciosl** [comment]
@matallo LGTM :+1: . I think I copied the wrong example, isn't it?

**Comment by juanignaciosl** [comment]
Retest this, please.

**Comment by matallo** [comment]
I added the title and there wasn't really a template for not success/error sooo, no prob

**Comment by josecruz** [comment]
After talk with @matallo we're going to remove separation lines to match the designs.

ty!

**Comment by matallo** [comment]
screen shot 2015-09-15 at 15 40 32

**Comment by josecruz** [comment]
awesomic :green_heart:

**Comment by juanignaciosl** [comment]
Retest this, please.

**Comment by juanignaciosl** [comment]
Retest this, please.

**Pull request merged**